poc_iot_injector/src/server.rs

```diff
@@ -99,36 +102,56 @@ impl Server {
 }
 
 async fn submit_txns(
-    txn_service: &mut TransactionService,
+    txn_service: Arc<Mutex<TransactionService>>,
     store: &mut FileStore,
-    keypair: &Keypair,
+    keypair: Arc<Keypair>,
     after_utc: DateTime<Utc>,
     before_utc: DateTime<Utc>,
 ) -> Result {
     let file_list = store
         .list_all(FileType::LoraValidPoc, after_utc, before_utc)
         .await?;
 
-    let mut stream = store.source(stream::iter(file_list).map(Ok).boxed());
-    let ts = before_utc.timestamp_millis();
+    let stream = store.source(stream::iter(file_list).map(Ok).boxed());
```

Comment:
If you really don't care in which order the files themselves  are processed either (separate from the ordering within the file) you could use `store.source_unordered` which will load files concurrently too

Reply:
Yeah, I don't think the order of processing matters as long as it is done for the batch after the `last_poc_submission_ts`. I changed it to `store.source_unordered` here 838418c